src/combination.ts

```diff
@@ -101,3 +124,21 @@ export const combineQueries = <Queries extends QueryObserverResult[]>(
     queries: queries as any,
   };
 };
+
+export const suspenseCombineQueries = <Queries extends QueryObserverResult[]>(
+  queries: [...Queries],
+): SuspenseCombinedQueriesResult<Queries> => {
+  const base = getBase(queries);
+
+  // Loading and Error states will be handled by suspense and error
+  // boundaries so unlike the non-suspense version we only need to
+  // account for the DefinedQueryObserverResult state
+  return {
+    ...base,
+    status: 'success',
```

Comment:
The main scenario for using the suspense helpers would be in places where we'd want to show a single loader/fallback for a specific section (and also one single error fallback) while not having to deal with "async data", as in, the data will always be there since the main view would only be rendered after queries have succeeded/failed, correct?

Reply:
yeah I'd say in general that is right - but even when you want a more custom loading state ui lower down you could just wrap that section in another boundary - I think there is also ways to control how long the app might not display any different ui until it takes some amount of time etc and maybe some hooks into that inbetween state 

I think its seen overall an improvement of the experience for the end user as it kind of combines all queries down stream - I am actually not 100% certain what the use case is for `useCombinedSuspense` queries and at first I didn't expect react-query to have it

Reply:
Maybe we could start simple and only include the simple `useSuspenseAPIQuery` (and maybe the infinite query if you think we'll be using it right away)? Just bringing it up so we keep the surface area as small as we can for now.

Reply:
I think I'd agree with @aecorredor here -- it's not clear to me what the value of "combining" suspending queries is, since there's no real async state. I'd suggest removing this for simplicity.

Reply:
@jkdowdle Got it, that makes sense (running parallel suspended queries) -- seems like a worthwhile use case to support.

Reply:
ok confirmed behavior. I put some artificial delay and logs in the query functions

without combined

```bash
console.log
    First waiting for 500...

  console.log
    done waiting

  console.log
    Second waiting for 250...

  console.log
    done waiting
```

vs

with combined

```bash
  console.log
    First waiting for 500...

  console.log
    Second waiting for 250...

  console.log
    done waiting

  console.log
    done waiting

```